bin/build-ts.mjs

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env node
+
+import { spawn } from 'node:child_process'
+import { once } from 'node:events'
+import { existsSync } from 'node:fs'
+import path from 'node:path'
+import { stdin, env } from 'node:process'
+import { createInterface as readLines } from 'node:readline'
+import { fileURLToPath } from 'node:url'
+
+const fromYarn = 'npm_execpath' in env
+const exe = fromYarn ? env.npm_execpath : 'corepack'
+const argv0 = fromYarn ? [] : ['yarn']
+
+const cwd = fileURLToPath(new URL('../', import.meta.url))
+
+for await (const line of readLines(stdin)) {
+  const { location, name } = JSON.parse(line)
+  if (existsSync(path.join(cwd, location, 'tsconfig.json'))) {
+    const cp = spawn(exe, [...argv0, 'tsc', '-p', location], {
+      stdio: 'inherit',
+      cwd,
+    })
+    await Promise.race([
+      once(cp, 'error').then(err => Promise.reject(err)),
+      await once(cp, 'exit')
+        .then(([code]) => (code && Promise.reject(new Error(`Non-zero exit code when building "${name}": ${code}`)))),
+    ])
```

Comment:
this could probable be simplified with `execa` to 

```js
await execa(exe, [...argv0, 'tsc', '-p', location], { cwd, stdio: 'inherit' });
```

Reply:
we already have the dependency and are using it other places in the code (devDependencies only so not really affecting bundle size or anything)

Reply:
I'd rather see us getting rid of it than relying on it more.

Reply:
> we should propose a new one

Yup that's a great idea. I found this issue https://github.com/nodejs/node/issues/38823 - and I see that you already worked on a for a bit! 💪

but I still think that until node.js has a good promise api for cp, it makes sense to use the (imo) next best thing which is a popular npm module. kind of like how we used bluebird before promises were a thing natively, or how I used [fs-extra](https://github.com/jprichardson/node-fs-extra) before [fs/promises](https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_promises_api) (which also lead to a very easy migration once native support was there)

> We're not helping anyone by using `execa`. What would you like to see exactly?

IMO we are helping ourselves by using execa, because it makes code easier to read and maintain, compared to always reinventing the wheel by implementing promise wrappers everywhere whenever we need to use child_process, which adds a lot of (sometimes a bit voodoo) code.

Alternatively I'm open to creating a promise-based cp wrapper inside uppy and use that instead of execa for when we need to make processes with promises, in order to keep our dependencies down. But I don't think duplicating promise wrapping logic everywhere makes our code easier to maintain than using execa does.

Reply:
Ultimately that doesn't really matter to me, if the majority thinks we should go with execa, I'll be fine with that. My preference would be that we don't, and if creating our own wrapper is an option that everyone can get behind, let's do that.
In my experience, it's not possible to make a `Promise` API that fulfils all the use cases (which is one of the reasons why Node.js still doesn't have a promise API), so there are some compromise to make one way or another. The upside of having explicit promise wrapper is that it's very clear what is happening; conversely, when you use an external library, you have to assume what is happening behind the scenes (or know their documentation by heart). Having our own wrapper is maybe the step in between.

Reply:
Even if it's not possible to make a promise api that fulfills all use cases, if it fulfills the most common use cases, then maybe it's still a good thing? kind of like what execa does. personally I haven't yet found any use cases where execa doesn't work, so I just keep using it. But if it doesn't work for some particular use case then one can just fall back to the normal child_process api without promises. maybe someone wants to make a promise-wrapper util? if not I think keeping execa is better until node.js has promise support.